<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>470</width>
    <height>453</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Repeat run</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="5" column="0" colspan="2">
    <widget class="QGroupBox" name="CPlot_group">
     <property name="title">
      <string>Cumulative plot</string>
     </property>
     <widget class="QCheckBox" name="CPlot_checkbox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>161</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Turn on cumulative plot</string>
      </property>
     </widget>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QGroupBox" name="sequencer_group">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>Sequencer</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Program</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="program_path_line_edit"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="choose_program_button">
          <property name="text">
           <string>Choose program</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="edit_program_button">
          <property name="text">
           <string>Edit program</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QGroupBox" name="config_file_group">
     <property name="title">
      <string>Config file</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="1" column="0">
       <widget class="QPushButton" name="reload_config_button">
        <property name="text">
         <string>Reload from config file</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="edit_config_button">
        <property name="text">
         <string>Edit config file</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="save_config_button">
        <property name="text">
         <string>Save current settings to config file</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="QLabel" name="config_file_label">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="repeat_group">
     <property name="title">
      <string>Repetitions</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="3">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Count:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Max repetition:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="indefinite_checkbox">
        <property name="text">
         <string>Repeat indefinitely</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QSpinBox" name="repetition_spinbox">
        <property name="maximum">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QLabel" name="current_run_label">
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Auto-realign</string>
     </property>
     <widget class="QSplitter" name="splitter">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>401</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QCheckBox" name="realign_checkbox">
       <property name="text">
        <string>Enable realign</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Realign stages</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="realign_count_spinbox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="status_display">
     <property name="text">
      <string>TextLabel</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QPushButton" name="stop_button">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QPushButton" name="repeat_button">
     <property name="text">
      <string>Repeat</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QGroupBox" name="save_data_group">
     <property name="title">
      <string>Save data</string>
     </property>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>431</width>
        <height>25</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QCheckBox" name="auto_save_checkbox">
         <property name="text">
          <string>save data automatically</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="export_data_button">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>To data_exported</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>repeat_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>repeat()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>186</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>choose_program_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>choose_program()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edit_program_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>edit_program()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reload_config_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>reload_config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_config_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>save_config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>392</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>export_data_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>export_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>432</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edit_config_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>edit_config()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>134</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stop_button</sender>
   <signal>clicked()</signal>
   <receiver>Form</receiver>
   <slot>stop_repetition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>134</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>repeat()</slot>
  <slot>reload_config()</slot>
  <slot>save_config()</slot>
  <slot>choose_program()</slot>
  <slot>edit_program()</slot>
  <slot>export_data()</slot>
  <slot>edit_config()</slot>
  <slot>stop_repetition()</slot>
 </slots>
</ui>
